core/src/query.rs

```diff
@@ -31,6 +38,87 @@ pub struct MongoQuery {
 }
 
 impl MongoQuery {
+    fn get_sql_query_namespaces(sql_query: &str, db: &String) -> Result<BTreeSet<Namespace>> {
+        let command = Command::new(GetNamespaces::new(sql_query.to_string(), db.to_string()));
+
```

Comment:
And building on my suggestion, we can get rid of these `Command::new` wrappers entirely!